### PR TITLE
BL-948 - UX: Build Underdog Devs UI Kit

### PR DESCRIPTION
**Pull request by albert-szabo**
## Description

To standardize design elements and provide reference UI materials, we created a basic UI kit across four pages in an updated Figma file.

We have included two rounds of tweaks based on feedback from our UX manager Swapnali. We also added updates based on feedback provided by our stakeholder Jason at the stakeholder meeting on 19 OCT 2022. All suggested improvements were implemented in Figma, resulting in the final pages, screenshots of which are included to be merged.

A brief description of each of the four pages follows. For additional information, please refer to the new README included with this pull request.

#### Colors / Palettes

This page includes six swatches to standardize colors, usage information based on contrast checks for accessibility, and three explanatory notes.

#### Typography / Fonts

This page includes the newly reduced two fonts (down from six previously) for the project, example text, example headers, and two explanatory notes.

#### Buttons

This page includes a self-serve / grab-and-go buttons table with 12 button varieties, color-matched standard buttons, and one explanatory note.

#### Grids

This page includes nine layout grids, two options for mobile layouts, and two explanatory notes.

#### Note on Viewing the PNG Files

Reviewers may have to open the second commit (from the top) individually under the "commits" tab to see the PNG files rendered correctly. Under the "files changed" tab, they may not display correctly since GitHub tries to render them as code blocks.

If that doesn't work, refreshing the page a few times (it can be temperamental) and trying to access it in Google Chrome might help.

#### Note on Accessing the Updated Figma File

Reviewers may not be able to immediately access the updated Figma file because both logging in to a Figma account and being granted specific access permission may be required, even for viewing. We are aware of this issue and the need to adjust access controls, which may only be available to our UX manager Swapnali.

However, this should not be a blocker for reviewing because the four screenshots included are accurate and up-to-date representations of all four relevant pages and can therefore be referenced instead.

### Figma Link

[Updated Figma File](https://www.figma.com/file/6x8Gl4Ac3borEHW6RrhJiG/Underdog-Devs-Ant-Design?node-id=4%3A190152)

### Video Link

[Loom Video](https://www.loom.com/share/86daf485fd874e1eb99ddf6e36055d4f)

### Ticket Link

[Jira Ticket (BL-948)](https://bloomtechlabs.atlassian.net/browse/BL-948?atlOrigin=eyJpIjoiNDE0NDEyNjVjM2I5NDI2MzgwOTNhZjUwMWVhNDdlYmYiLCJwIjoiaiJ9)

## Checklist

- [x] My design/diagram follows the style guidelines of this project
- [x] My docs are organized within an appropriate folder structure in our repo
- [x] Pull request description clearly describes docs made & motivations for said additions

### CODE REVIEW

**Review by zoebrantley — APPROVED** [review]
Really love having all of this standardized. Love the colors chosen and overall layout. Well done.

**Comment by JTCoates45** [comment]
Having this all in one spot and easily accessible is awesome!

**Review by JamesFincher — APPROVED** [review]
This is amazing work. It's clean, well documented, and will allow us to make the ui look very nice!

File: UI-Kit/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+To provide reference design materials and standardize elements across the app, we created a basic UI kit initially consisting of four pages in an updated Figma file.
+
+The four pages are palettes (colors), typography (fonts), buttons, and grids.
+
+### Colors / Palettes
+
+<img title="Colors-Palettes" alt="A Figma page with a row of six colors swatches and a white box with usage information for each color below" src="./Colors-Palettes.png">
+
+Here's the colors / palettes page. We added six color swatches matched to the exact colors from the current UD marketing website, with one exception. The primary UD logo orange did not pass all contrast tests for accessibility, so we created a new darker shade labeled "high-contrast" that has sufficient contrast for all use cases.
+
```
Comment (joanterm) [inline UI-Kit/README.md:12]: This particular palette is a great choice, and I like how organized everything is.

Comment (roxannebobby) [reply UI-Kit/README.md:12]: I love the colors! Well contrasting and fit the purpose.

File: UI-Kit/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+We included hexadecimal color codes for easy reference, and usage information for each color specifying which use cases are appropriate based on our contrast tests for accessibility. We also included three helpful explanatory notes to add context.
+
+#### Note on Colors for Text
+
+Although most standard (small / body) text is expected to be black, we included usage information for both small and normal text in our descriptions to account for the case of the six colors being used for some other type of text, perhaps as section headers or page titles.
+
```
Comment (roxannebobby) [inline UI-Kit/README.md:18]: Great advance thinking :)

File: UI-Kit/README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+Here's the buttons page. We added a self-serve table where teammates can grab the button(s) they need by combining the category (one of the four columns) and the state / type (one of the three rows). We also included an explanatory note reminding teammates to add alt text for accessibility when using the small square buttons.
+
+The blue color swatch in the upper left corner of the table is there to signify that the two relevant standard (primary) category buttons (the active and hover / pressed variants) have already been color matched to the exact brighter blue specified on the colors page. They do not need to have their fill color modified, and can be used as-is.
+
+### Grids
+
+<img title="Grids" alt="A Figma page with nine layout grids for devices of various sizes in both portrait and landscape orientations showing broad vertical bars and translucent rectangles, with small orange logos in their upper left corners, and three explanatory notes below them" src="./Grids.png">
+
```
Comment (roxannebobby) [inline UI-Kit/README.md:38]: This is impressive and very professional!

**Review by roxannebobby — APPROVED** [review]
So very professional well done!

**Review by randynobx — APPROVED** [review]
Great work, I love the look and 'feel' of what you have come up with!

**Review by tim-sansone — APPROVED** [review]
This entire set of documents is extraordinarily well crafted and professional. The content is good and the presentation is very helpful. I particularly like the color palettes page where you give specific instructions for the use cases of each color. This is extremely useful for any developer who may not have much natural intuition for making design decisions.

I had to work extremely hard to find anything to criticize on this PR because this work is so good! But, I was up for the challenge. You spelled the word "square" wrong in your note at the bottom of the buttons page. :)

**Review by nikolayglushetskiy — APPROVED** [review]
Awesome!

**Comment by albert-szabo** [comment]
Thanks for the good catch, @tim-sansone! The typo has been fixed in Figma.

**Review by mina-mikhael — APPROVED** [review]
Awesome Job, I like how everything is organized and explained with details and visual examples

**Review by JeffDesch — APPROVED** [review]
Excellent work, the unifying color scheme will give even the existing frontend parts a polished look. Especially notable is the good work on the status clarity of the buttons, something I've seen even professional sites sometimes miss.

**Comment by NelsonMaldonado** [comment]
Overall excellent job, this new UI Kit gives the team the tools to make our code easier to implement. Thanks to the new standardized ant design buttons my team and I were able to refactor the nav bar section and got rid of pages of unnecessary code. The loom video is also very useful to get an understanding on how to use this new Ant Design UI Kit.

**Review by ashurik522 — APPROVED** [review]
Very clean and professionally made. It is gonna look great!

**Review by TophertheCoder — APPROVED** [review]
These are incredibly detailed and thorough! Well put together without going overboard on details. Great job!

**Pull request merged**